### PR TITLE
TST: groupby.first/last retains categorical dtype

### LINKED ISSUE #33090: SeriesGroupBy.first / last loses categorical dtype
On 1.0.3 and master:
```python
import pandas as pd

df = pd.DataFrame({"a": [1, 2, 3]})
df["b"] = df["a"].astype("category")

print(df.groupby("a")["b"].first())
print(df.groupby("a")["b"].last())
```
gives
```python
a
1    1
2    2
3    3
Name: b, dtype: int64
a
1    1
2    2
3    3
Name: b, dtype: int64
```
but the dtype should still be categorical and not int64. This seemingly wrong output is explicitly tested for here: https://github.com/pandas-dev/pandas/blob/master/pandas/tests/groupby/aggregate/test_aggregate.py#L461

### PR DESCRIPTION
**Pull request by debnathshoham**
- [x] closes #33090
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/groupby/test_categorical.py [hunk 1]
```diff
@@ -1741,3 +1741,15 @@ def test_groupby_categorical_indices_unused_categories():
     assert result.keys() == expected.keys()
     for key in result.keys():
         tm.assert_numpy_array_equal(result[key], expected[key])
+
+
+def test_groupby_last_first_preserve_categoricaldtype():
```
Comment (jreback) [inline pandas/tests/groupby/test_categorical.py:1746]: can you parameterize over first/last

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/groupby/test_categorical.py (hunk 1 above)
Comment (debnathshoham) [reply pandas/tests/groupby/test_categorical.py]: Done

**Review by phofl — CHANGES_REQUESTED** [review]

**Review by phofl — APPROVED** [review]
Lgtm

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @debnathshoham 

since we get a lot of PRs, periodically rebasing yours is a good idea as it moves them to the front of the queue (assuming they are ready to go)

**Pull request merged**